File: Lab 1/Lan-Java/CaesarCipher.java
import java.util.Scanner;

public class CaesarCipher {
    public static String encrypt(String text, int shift) {
        StringBuilder result = new StringBuilder();
        for (char character : text.toCharArray()) {
            if (Character.isLetter(character)) {
                char base = Character.isLowerCase(character) ? 'a' : 'A';
                result.append((char) ((character - base + shift) % 26 + base));
            } else {
                result.append(character);
            }
        }
        return result.toString();
    }

    public static String decrypt(String cipher, int shift) {
        StringBuilder result = new StringBuilder();
        for (char character : cipher.toCharArray()) {
            if (Character.isLetter(character)) {
                char base = Character.isLowerCase(character) ? 'a' : 'A';
                result.append((char) ((character - base - shift + 26) % 26 + base));
            } else {
                result.append(character);
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int shift = 3;

        System.out.println("Select an option:");
        System.out.println("1. Encrypt a message");
        System.out.println("2. Decrypt a message");
        int choice = sc.nextInt();
        sc.nextLine();

        if (choice == 1) {
            System.out.println("Enter the text to be encrypted:");
            String text = sc.nextLine();
            String encrypted = encrypt(text, shift);
            System.out.println("Encrypted Text: " + encrypted);
        } else if (choice == 2) {
            System.out.println("Enter the text to be decrypted:");
            String cipher = sc.nextLine();
            String decrypted = decrypt(cipher, shift);
            System.out.println("Decrypted Text: " + decrypted);
        } else {
            System.out.println("Invalid option. Please select 1 or 2.");
        }

        sc.close();
    }
}

File: Lab 1/Lan-Python/CaesarCipher.py
# def encrypt(text, shift):
#     result = ""
#     for char in text:
#         if char.isalpha():
#             # Determine if the character is uppercase or lowercase
#             base = ord('a') if char.islower() else ord('A')
#             # Apply the Caesar cipher formula
#             result += chr((ord(char) - base + shift) % 26 + base)
#         else:
#             # Keep non-alphabetic characters unchanged
#             result += char
#     return result
#
#
# def decrypt(cipher, shift):
#     result = ""
#     for char in cipher:
#         if char.isalpha():
#             # Determine if the character is uppercase or lowercase
#             base = ord('a') if char.islower() else ord('A')
#             # Apply the reverse Caesar cipher formula
#             result += chr((ord(char) - base - shift + 26) % 26 + base)
#         else:
#             # Keep non-alphabetic characters unchanged
#             result += char
#     return result
#
#
# def main():
#     shift = 3
#
#     print("Select an option:")
#     print("1. Encrypt a message")
#     print("2. Decrypt a message")
#
#     try:
#         choice = int(input())
#
#         if choice == 1:
#             print("Enter the text to be encrypted:")
#             text = input()
#             encrypted = encrypt(text, shift)
#             print("Encrypted Text:", encrypted)
#         elif choice == 2:
#             print("Enter the text to be decrypted:")
#             cipher = input()
#             decrypted = decrypt(cipher, shift)
#             print("Decrypted Text:", decrypted)
#         else:
#             print("Invalid option. Please select 1 or 2.")
#     except ValueError:
#         print("Please enter a valid number.")
#
#
# if __name__ == "__main__":
#     main()


def caesar(text, shift, mode='encrypt'):
    result = ""
    for char in text:
        if char.isalpha():
            base = ord('a') if char.islower() else ord('A')
            # If decrypting, reverse the shift
            s = shift if mode == 'encrypt' else -shift
            result += chr((ord(char) - base + s) % 26 + base)
        else:
            result += char
    return result


if __name__ == "__main__":
    shift = 3

    print("Select an option:\n1. Encrypt a message\n2. Decrypt a message")
    choice = int(input())

    if choice in [1, 2]:
        mode = 'encrypt' if choice == 1 else 'decrypt'
        text = input(f"Enter the text to be {mode}ed:\n")
        result = caesar(text, shift, mode)
        print(f"{mode.capitalize()}ed Text:", result)
    else:
        print("Invalid option. Please select 1 or 2.")
